core/src/dht/dht_store.rs

```diff
@@ -58,12 +58,16 @@ impl DhtStore {
         address: Address,
         tag: String,
     ) -> Result<BTreeSet<EntityAttributeValueIndex>, HolochainError> {
-        self.meta_storage.read()?.fetch_eavi(
-            Some(address),
-            Some(format!("link__{}", tag)),
-            None,
-            IndexQuery::default(),
-        )
+        let index_query = IndexQuery::new_only_prefixes(vec!["link__", "removed_link__"]);
+        let filtered =
+            self.meta_storage
+                .read()?
+                .fetch_eavi(Some(address), Some(tag), None, index_query)?;
+
+        Ok(filtered
+            .into_iter()
+            .filter(|eav| eav.attribute().starts_with("link__"))
+            .collect())
```

Comment:
I don't understand why this does or should work. Seems like you are just ignoring the removals.. ??

Reply:
BTW, we clarified this per voice-chat yesterday. So `fetch_eavi` only returns the latest version of potentially multiple EAVs that match the query. With the prefixes set in line 61, all EAVs where the attribute matches either `link__<tag>` or `removed_link_<tag>` are considered the same. If we get the latest of these each, it will be either a `link__<tag>` or `removed_link_<tag>`, depending on if the last action (for this specific link!) was an add or removal. So yes, if remove all those links for which the latest action is a removal we get exactly what we want.

And, this could get more documented and tested :)